app/presenters/tree_builder_ops_features.rb

```diff
@@ -0,0 +1,249 @@
+class TreeBuilderOpsFeatures < TreeBuilder
+  include CompressedIds
+
+  has_kids_for Hash, [:x_get_tree_hash_kids]
+
+  attr_reader :role, :features, :editable
+
+  def initialize(name, type, sandbox, build, role:, features:, editable: false)
+    @role     = role
+    @editable = editable
+    @features = features
+
+    super(name, type, sandbox, build)
+  end
+
+  private
+
+  def feature_tree
+    recurse_menu_items(filtered_menus, root_select)
+  end
+
+  def filtered_menus
+    menus = []
+
+    Menu::Manager.each do |section|
+      next if section.id == :cons && !Settings.product.consumption
+      next if section.name.nil?
+      next unless Vmdb::PermissionStores.instance.can?(section.id)
+
+      menus.push(section)
+    end
+
+    menus
+  end
+
+  def recurse_menu_items(items, checked = false)
+    items.map do |item|
+      case item
+      when Menu::Section
+        menu_section_to_node(item)
+      when Menu::Item
+        if item.feature.nil? || !MiqProductFeature.feature_exists?(item.feature)
+          next
+        end
+        menu_item_to_node(item, checked)
+      when String
+        feature_name_to_node(item, checked)
+      else
+        item
+      end
+    end
+  end
+
+  def menu_section_to_node(section)
+    node = default_node.merge!(
+      :id      => "#{node_id}___tab_#{section.id}",
+      :text    => _(section.name),
+      :tooltip => _("%{title} Main Tab") % {:title => section.name},
+      :data    => {}
+    )
+
+    node[:data][:kids] = recurse_menu_items(section.items).compact
```

Comment:
can we please not do this? There is no reason this tree should be this special, is there? Each *_tree_kids method should only return that 1 node, and not include any children - the tree should be constructed from the top..